src/structure.ts

```diff
@@ -81,44 +81,48 @@ const isCommand = (value: any, warn = false): value is Command => {
     return true;
 };
 
+type AccType = { [key: string]: Command };
+
 /**
  * Validate that a set of values are {@link Command} objects
  */
-export const validateCommands = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmds: any[], warn = false) =>
-    cmds.reduce((acc, cmd) => {
-        if (!isCommand(cmd, warn)) return acc;
+export const validateCommands = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmds: unknown[], warn = false) =>
+cmds.reduce((acc: AccType, cmd: unknown) => {
+    if (!isCommand(cmd as Command, warn)) return acc;
 
-        // Check the command doesn't already exist
-        if (acc[cmd.name]) {
-            if (warn)
-                console.warn(`Command ${cmd.name} already exists`);
-            return acc;
-        }
+    const command = cmd as Command;
 
-        // Add the command
-        return {
-            ...acc,
-            [cmd.name]: cmd,
-        };
-    }, {}) as Commands<Ctx, Req, Sentry>;
+    // Check the command doesn't already exist
+    if (acc[command.name]) {
+        if (warn)
+            console.warn(`Command ${command.name} already exists`);
+        return acc;
+    }
+
+    // Add the command
+    return {
+        ...acc,
+        [command.name]: command,
+    };
+}, {} as AccType) as Commands<Ctx, Req, Sentry>;
 
 /**
  * Validate that a given value is a {@link Component} object
  */
-const isComponent = (value: any, warn = false): value is Component => {
+const isComponent = (value: unknown, warn = false): value is Component => {
     if (typeof value !== 'object' || value === null) {
         if (warn)
             console.warn('Expected component to be an object');
         return false;
     }
 
-    if (typeof value.name !== 'string' || !value.name.length) {
+    if (typeof value === 'object' && value !== null && 'name' in value && typeof value.name === 'string' && value.name.length > 0) {
```

Comment:
We've already checked that we have an object that isn't null above, we do not need to check again here.

You've also completely changed what this logic is doing -- this if statement should capture anything that isn't a string, or doesn't have a length -- instead, you've now got it capturing anything that is a string that does have a length...